src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -0,0 +1,230 @@
+                    if (rc == JFileChooser.APPROVE_OPTION) {
+                        importOpenApiDefinition(chooser.getSelectedFile());
+                    }
+
+                }
+            });
+        }
+        return menuImportLocalOpenApi;
+    }
+
+    /* Menu option to import a OpenApi file from a given URL. */
+    private ZapMenuItem getMenuImportUrlOpenApi() {
+        if (menuImportUrlOpenApi == null) {
+            menuImportUrlOpenApi = new ZapMenuItem("openapi.topmenu.tools.importRemoteOpenApi");
+            menuImportUrlOpenApi
+                    .setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importRemoteOpenApi.tooltip"));
+
+            final ExtensionOpenApi shadowCopy = this;
+            menuImportUrlOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    SwingUtilities.invokeLater(new Runnable() {
+
+                        @Override
+                        public void run() {
+                            new ImportFromUrlDialog(View.getSingleton().getMainFrame(), shadowCopy);
+                        }
+                    });
+                }
+            });
+        }
+        return menuImportUrlOpenApi;
+    }
+
+    public void importOpenApiDefinition(final URI uri) {
+        Thread t = new Thread(THREAD_PREFIX + threadId++) {
+
+            @Override
+            public void run() {
+                Requestor requestor = new Requestor(HttpSender.MANUAL_REQUEST_INITIATOR);
+                requestor.addListener(new HistoryPersister());
+                try {
+                    importOpenApiDefinition(requestor.getResponseBody(uri));
+                } catch (SwaggerException e) {
+                    if (View.isInitialised()) {
+                        View.getSingleton().showWarningDialog(Constant.messages.getString("openapi.parse.error"));
+                    }
+                    LOG.error(e.getMessage(), e);
+                } catch (IOException e) {
+                    if (View.isInitialised()) {
+                        View.getSingleton().showWarningDialog(Constant.messages.getString("openapi.io.error"));
+                    }
+                    LOG.error(e.getMessage(), e);
+                } catch (Exception e) {
+                    LOG.error(e.getMessage(), e);
+                }
+            }
+
+        };
+        t.start();
+    }
+
+    public void importOpenApiDefinition(final File file) {
+        Thread t = new Thread(THREAD_PREFIX + threadId++) {
+
+            @Override
+            public void run() {
+                try {
+                    importOpenApiDefinition(FileUtils.readFileToString(file));
+                } catch (SwaggerException e) {
+                    if (View.isInitialised()) {
+                        View.getSingleton().showWarningDialog(Constant.messages.getString("openapi.parse.error"));
```

Comment:
Could the methods (`importOpenApiDefinition` and `importOpenApiDefinition`) be refactored to avoid this duplication?